tests/common/auth_test.go

```diff
@@ -599,16 +599,33 @@ func TestAuthMemberRemove(t *testing.T) {
 		rootAuthClient := testutils.MustClient(clus.Client(WithAuth(rootUserName, rootPassword)))
 		testUserAuthClient := testutils.MustClient(clus.Client(WithAuth(testUserName, testPassword)))
 
-		memberId, clusterId := memberToRemove(ctx, t, rootAuthClient, clusterSize)
+		memberId, _ := memberToRemove(ctx, t, rootAuthClient, clusterSize)
 
 		// ordinary user cannot remove a member
 		_, err := testUserAuthClient.MemberRemove(ctx, memberId)
 		require.ErrorContains(t, err, PermissionDenied)
 
 		// root can remove a member
-		removeResp, err := rootAuthClient.MemberRemove(ctx, memberId)
-		require.NoError(t, err, "MemberRemove failed")
-		require.Equal(t, removeResp.Header.ClusterId, clusterId)
+		// client can connect to removed member and return "etcdserver: server stopped"
+		rootAuthClient.MemberRemove(ctx, memberId)
+		require.Eventually(t, func() (done bool) {
+			// linearizable member list
+			resp, err := rootAuthClient.MemberList(ctx, false)
+			if err != nil {
+				// client may connect to removed member
+				// client retry interceptor won't retry if it's context deadline exceeded
+				return false
+			}
+			found := false
+			for _, member := range resp.Members {
+				if member.ID == memberId {
+					found = true
+					break
+				}
+			}
+			require.False(t, found, "expect removed member not found in member list")
+			return true
+		}, time.Second*5, time.Millisecond*100)
```

Comment:
This looks like a workaround for a fact that you cannot remove member's endpoint from client.
Best option would be for etcdctl to support `SetEndpoints` like `clientv3.Client`.

Reply:
It turns out integration tests use `listen-grpc-url` as endpoints but member list clientURLs are configured with local unix listener with unique incrementing port. 

So there is no easy way to exclude the removed member endpoint from the original `clus.Endpoints()` unless big refactoring of the test framework.